src/architect/project/project.utils.ts

```diff
@@ -0,0 +1,290 @@
+    // download any selection that has a homepage
+    for (const selection of Object.values(selections)) {
+      if (selection.architect_yml && selection.architect_yml.homepage) {
+        const url = selection.architect_yml.homepage.replace('https://github.com/', '');
+        await this.downloadRepo(url, project_dir + '/' + selection.name.toLowerCase());
+      }
+    }
+    await new Promise(resolve => setTimeout(resolve, 2000));
+  }
+
+  static async createArchitectYaml(selections: Dictionary<any>, project_dir: string): Promise<void> {
+    // get actual service name for depends_on
+    for (const selection of Object.values(selections)) {
+      if (selection.depends_on) {
+        const depends_on_proj = selections[selection.depends_on];
+        const services = depends_on_proj.architect_yml.services;
+        if (Object.keys(services).length === 1) {
+          selection.depends_on = Object.keys(services)[0];
+        } else {
+          const service_name = Object.keys(services).find((name: any) => !services[name].image);
+          selection.depends_on = service_name;
+        }
+      }
+    }
+
+    const combined_yml: Dictionary<any> = {};
+    combined_yml['name'] = project_dir.replace('_', '-');
+    const ignored = ['name', 'description', 'homepage', 'keywords'];
+    for (const selection of Object.values(selections)) {
+      for (const [key, val] of Object.entries(selection.architect_yml as ComponentSpec)) {
+        if (ignored.includes(key)) {
+          continue;
+        }
+
+        if (key !== 'services') {
+          combined_yml[key] = combined_yml[key] ? { ...combined_yml[key], ...val } : val;
+          continue;
+        }
+
+        const yml_services = val as Dictionary<ServiceSpec>;
+        const service_keys = Object.keys(yml_services);
+        if (service_keys.length === 1) {
+          const service_key = service_keys[0];
+          const service = yml_services[service_key];
+          if (service.build) {
+            service.build['context'] = './' + selection.name;
+          }
+          if (selection.depends_on) {
+            yml_services[service_key]['depends_on'] = [selection.depends_on];
+          }
+          combined_yml[key] = combined_yml[key] ? { ...combined_yml[key], ...yml_services } : yml_services;
+        } else {
+          for (const [service_name, service] of Object.entries(yml_services)) {
+            if (service.build && !service.image) {
+              combined_yml[key] = combined_yml[key] ? { ...combined_yml[key], ...{ [service_name]: service } } : { [service_name]: service };
+              service.build['context'] = './' + selection.name;
+              if (selection.depends_on) {
+                combined_yml[key][service_name]['depends_on'] = [selection.depends_on];
+              }
+            }
+          }
+        }
+      }
+    }
+
+    const app_key = Object.keys(combined_yml['interfaces']).find(key => key === 'app');
```

Comment:
Is this because multiple starter projects use the same interface name?

Reply:
No, it is because they use different interface names. When the interfaces combine, we don't know which to choose. So, I made it look for the interface named `app`. If `app` doesn't exist, then just pick the first one interface.